src/Database/LSMTree/Internal/UniqCounter.hs

```diff
@@ -23,16 +24,17 @@ uniqueToRunNumber :: Unique -> RunNumber
 uniqueToRunNumber = coerce
 
 -- | An atomic counter for producing 'Unique' values.
---
--- TODO: could we use a PrimVar here?
-newtype UniqCounter m = UniqCounter (StrictMVar m Word64)
+newtype UniqCounter m = UniqCounter (PrimVar (PrimState m) Word64)
```

Comment:
```suggestion
newtype UniqCounter s = UniqCounter (PrimVar s Word64)
```

Reply:
Changing `(PrimState m)` to `s` here causes a lot of "upstream" breakages in the type-class constraint which call functions that interact with `UniqCounter`. Generally it requires creating a new type variable `s` in addition to the existing `m`, and adding `PrimState m ~ s` in a lot of places.

Is there a technical reason to make this change? Performance or memory implications I might not be aware of?

Reply:
It's not incorrect, it's just more restrictive to require `m` instead of `s`. We should add at least a TODO to change it, even if it would cause code changes elsewhere